src/plugins/intel_gpu/src/plugin/program.cpp

```diff
@@ -56,43 +57,74 @@ void Program::ValidateInputs(const std::shared_ptr<ngraph::Node>& op, std::vecto
                        << " op::v" << op->get_type_info().version << ")";
 }
 
-bool Program::CanProcessDynBatch(std::vector<std::shared_ptr<ngraph::Node>> ops, InferenceEngine::InputsDataMap networkInputs) const {
-    if (networkInputs.empty())
-        return false;
-
-    for (auto op : ops) {
-        // TODO: do we have any other exception cases?
-        if (std::dynamic_pointer_cast<ngraph::op::v1::Reshape>(op)) {
-            if (op->get_input_shape(0)[0] == op->get_output_shape(0)[0])
-                continue;
-        }
-
-        // List of the operations which can lead to invalid dynamic batch processing
-        if (std::dynamic_pointer_cast<ngraph::op::internal::NonMaxSuppressionIEInternal>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v5::NonMaxSuppression>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v4::NonMaxSuppression>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v3::NonMaxSuppression>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v1::NonMaxSuppression>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::PSROIPooling>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::ROIPooling>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::PriorBox>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::DetectionOutput>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v1::Reshape>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::Squeeze>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::Unsqueeze>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v1::Transpose>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v4::Proposal>(op) ||
-            std::dynamic_pointer_cast<ngraph::op::v0::Proposal>(op)) {
+auto getParamName = [](const std::shared_ptr<ngraph::Node> param) -> std::string {
+    const auto& names = param->get_output_tensor(0).get_names();
+    if (!names.empty())
```

Comment:
Why do you need this mix of node/tensor names?

Reply:
Because this code is also used from within QueryNetwork(), which is old-style based.
Rewriting all this stuff as well is beyond of this feature scope.

Reply:
@vladimir-paramuzov @mletavin I mean, is it ok that here we have a mix of new and old names?
we are using `get_output_tensor(0).get_names()` in new API where we work with tensor names, `get_friendly_names()` we are using for old API. Here we have a mix of old and new approaches

Reply:
The problem with the names is that in different situations, some of them are just empty (at least, in our test cases).
For example, in this given case, node and parameter input/output names are empty, while tensor names are OK.
So, I had no other way than choosing the name source depending on situation to make tests run.